locustfiles/all.py

```diff
@@ -0,0 +1,136 @@
+# -*- coding: utf-8 -*-
+""" Locust test for all APIs """
+from locust import HttpUser, between, events
+
+from utils.hosts import MilMoveHostMixin, MilMoveDomain, clean_milmove_host_users
+from utils.parsers import GHCAPIParser, PrimeAPIParser, SupportAPIParser
+from tasks import PrimeTasks, SupportTasks, ServicesCounselorTasks, TOOTasks
+
+# from flask import make_response, redirect
+from gevent.pool import Group
+
+
+# init these classes just once because we don't need to parse the API over and over:
+prime_api = PrimeAPIParser()
+support_api = SupportAPIParser()
+ghc_api = GHCAPIParser()
+
+
+class PrimeUser(MilMoveHostMixin, HttpUser):
+    """
+    Tests the Prime API.
+    """
+
+    # These attributes are used in MilMoveHostMixin to set up the proper hostname for any MilMove environment:
+    local_port = "9443"
+    domain = MilMoveDomain.PRIME  # the base domain for the host
+    is_api = True  # if True, uses the api base domain in deployed environments
+
+    # This attribute is used for generating fake requests when hitting the Prime API:
+    parser = prime_api
+
+    # These are locust HttpUser attributes that help define and shape the load test:
+    wait_time = between(0.25, 9)  # the time period to wait in between tasks (in seconds, accepts decimals and 0)
+    tasks = {PrimeTasks: 1}  # the set of tasks to be executed and their relative weight
+
+
+class SupportUser(MilMoveHostMixin, HttpUser):
+    """
+    Tests the Support API.
+    """
+
+    local_port = "9443"
+    domain = MilMoveDomain.PRIME
+    is_api = True
+
+    parser = support_api
+
+    wait_time = between(0.25, 9)
+    tasks = {SupportTasks: 1}
+
+
+class ServicesCounselorUser(MilMoveHostMixin, HttpUser):
+    """
+    Tests the MilMove Office app with the Services Counselor role.
+    """
+
+    local_protocol = "http"
+    local_port = "8080"
+    domain = MilMoveDomain.OFFICE
+
+    # This attribute is used for generating fake requests when hitting the GHC API:
+    parser = ghc_api
+
+    wait_time = between(0.25, 9)
+    tasks = {ServicesCounselorTasks: 1}
+
+
+class TOOUser(MilMoveHostMixin, HttpUser):
+    """
+    Tests the MilMove Office app with the TOO role.
+    """
+
+    local_protocol = "http"
+    local_port = "8080"
+    domain = MilMoveDomain.OFFICE
+
+    # This attribute is used for generating fake requests when hitting the GHC API:
+    parser = ghc_api
+
+    wait_time = between(0.25, 9)
+    tasks = {TOOTasks: 1}
```

Comment:
That's my vote. We can have one fall to run all the tests and use the UI to disable the users if we don't want to run any particular user.

I guess this is kind of an experiment so I'd be ok with leaving the duplication for now, but if you all would prefer to go ahead and consolidate, I'm fine with that, too

Reply:
I'm fine with leaving the duplication since we're experimenting, but can you add `TODO`'s for the ones that we should revisit deleting? Essentially I just want to make sure that a future person that sees both sees something in the codebase indicating we knowingly put in the duplication and that the plan is to likely delete after testing it for a while.

Reply:
I'm fine with leaving the duplication since we're experimenting, but can you add `TODO`'s for the ones that we should revisit deleting? Essentially I just want to make sure that a future person that sees both sees something in the codebase indicating we knowingly put in the duplication and that the plan is to likely delete after testing it for a while.

Reply:
Awesome, thank you!